### ISSUE TITLE
CLI flag to choose whether to open a new lighthouse page or not

### ISSUE THREAD

**Issue by Shivam93937** [GoogleChrome/lighthouse#16259]
**Feature request summary**

A CLI flag (something like `--reuse-page`) to re-use the existing page that was running on the browser that puppeteer connects to.
The page can be selected from `await lhBrowser.pages()`.

Yes

**What is the motivation or use case for changing this?**
It will help run lighthouse programmatically inside applications like CEF and Electron where opening a new page (by default) causes a failure when connecting to it through remote debugging port.

**How is this beneficial to Lighthouse?**
It will widen the audience to which lighthouse can be used and increase the tool's users.

Mentioned the changes in the screenshot:
![lh module changes](https://github.com/user-attachments/assets/bb4ecd9c-6a04-4d20-9b7d-a578c4a57e2b)

**Comment by adamraine** [comment]
This should be possible already with the Node API: https://github.com/GoogleChrome/lighthouse/blob/main/docs/puppeteer.md#option-1-launch-chrome-with-puppeteer-and-handoff-to-lighthouse

Is there a use case where you need to use the CLI and can't use the Node API?

**Comment by Shivam93937** [comment]
@adamraine I need to use the functionalities provided by lighthouse CI, (is-median runs, multiple runs, etc.) I cannot use node API if I want to use LIghthouse CI.

**Comment by adamraine** [comment]
Gotcha. I think a solution in Lighthouse CLI could be tricky because the CLI can still connect to an existing browser instance with multiple tabs open. Maybe a quick solution would be to re-use a page if it's the only page open.

**Comment by Shivam93937** [comment]
Yes agreed, this is the solution i proposed and I need a CLI flag for this, or I will have to fork and maintain the mono repo lhci and lighthouse too.

**Comment by Shivam93937** [comment]
@adamraine 
Is there any update on this discussion?
Should I provide a CL which can do the following, we will require changes both in lhci and lighthouse modules.
Adding this functionality will allow lighthouse to be run on Chromium Embedded Framework. 
I think the best way is to pass a CLI Flag, if we want to limit it to Single Page only, or execute a new page.

**Comment by Shivam93937** [comment]
@brendankenny Any opinions from your side?

**Comment by adamraine** [comment]
We are going to accept this proposal, add a `--reuse-page` or something.

Ideally, we could modify lighthouse CI to allow testing via the Node API so that a specific puppeteer page can be used, but that will be a much larger change than what is proposed here.

**Comment by Shivam93937** [comment]
![Image](https://github.com/user-attachments/assets/2bd441e3-1a9b-4362-bd4b-7097ef5080e6)

These changes in the lhci module does it for me and I am able to run lhci and also integrate a puppeteer script.

**Comment by Shivam93937** [comment]
Also, do we have a timeline for this feature?

**Comment by Shivam93937** [comment]
Hi @brendankenny @adamraine,
Do you need me to to do a PR and make the needed changes?
I have them ready for both lighthouse and lhci.
I can also share the patch files.

**Comment by adamraine** [comment]
A PR would be welcome if you have the changes ready to go :)
